pkg/storage/tsdb/config.go

```diff
@@ -357,8 +357,9 @@ type BucketStoreConfig struct {
 	SeriesHashCacheMaxBytes uint64 `yaml:"series_hash_cache_max_size_bytes" category:"advanced"`
 
 	// Controls whether index-header lazy loading is enabled.
-	IndexHeaderLazyLoadingEnabled     bool          `yaml:"index_header_lazy_loading_enabled" category:"advanced"`
-	IndexHeaderLazyLoadingIdleTimeout time.Duration `yaml:"index_header_lazy_loading_idle_timeout" category:"advanced"`
+	IndexHeaderLazyLoadingEnabled         bool          `yaml:"index_header_lazy_loading_enabled" category:"advanced"`
+	IndexHeaderLazyLoadingIdleTimeout     time.Duration `yaml:"index_header_lazy_loading_idle_timeout" category:"advanced"`
+	IndexHeaderEagerLoadingStartupEnabled bool          `yaml:"index_header_eager_loading_startup_enabled" category:"experimental"`
```

Comment:
Would it make sense to move these options to `indexheader.Config`? IMO it's clearer to have `IndexHeader.LazyLoadingEnabled`, and would also reduce the number of args in some calls, etc.

Reply:
I moved IndexHeaderEagerLoadingStartupEnabled to indexheader.Config. But I will move other configs in a separate PR.